packages/react-log-viewer/src/LogViewer/LogViewerToolbar.tsx

```diff
@@ -0,0 +1,108 @@
+  hasToolbar?: boolean;
+  customControls?: React.ReactNode | null;
+  searchedWordIndexes: number[];
+  rowInFocus: number;
+  searchedInput: string;
+  foundWordIndex: number;
+  scrollToRow: (searchedRow: number) => void;
+  setRowInFocus: (index: number) => void;
+  setSearchedInput: (input: string) => void;
+  setFoundWordIndex: (index: number) => void;
+  setSearchedWordIndexes: (indexes: number[]) => void;
+}
+
+export const LogViewerToolbar: React.FunctionComponent<LoggerToolbarProps> = ({
+  scrollToRow,
+  customControls = null,
+  hasToolbar = true,
+  rowInFocus,
+  searchedInput,
+  searchedWordIndexes,
+  foundWordIndex,
+  setRowInFocus,
+  setSearchedInput,
+  setSearchedWordIndexes,
+  setFoundWordIndex
+}) => {
+  const [currentSearchItemCount, setCurrentSearchItemCount] = useState<number>(1);
+  const DEFAULT_FOCUS = -1;
+  const DEFAULT_INDEX = 0;
+
+  /* Defaulting the first focused row that contain searched keywords */
+  useEffect(() => {
+    if (searchedWordIndexes.length >= 1) {
+      setFoundWordIndex(DEFAULT_INDEX);
+    }
+  }, [searchedWordIndexes]);
+
+  /* Clearing out the search input */
+  const handleClear = (): void => {
+    setSearchedInput('');
+    setFoundWordIndex(0);
+    setSearchedWordIndexes([]);
+    setRowInFocus(DEFAULT_FOCUS);
+  };
+
+  /* Moving focus over to next row containing searched word */
+  const handleNextSearchItem = (): void => {
+    const oldIndex = searchedWordIndexes.indexOf(rowInFocus);
+    const adjustedSearchCount = currentSearchItemCount + 1;
+    let temp = foundWordIndex;
+
+    if (oldIndex >= searchedWordIndexes.length - NUMBER_INDEX_DELTA) {
+      return null;
+    }
+
+    setCurrentSearchItemCount(adjustedSearchCount);
+    setFoundWordIndex(++temp);
+    scrollToRow(searchedWordIndexes[foundWordIndex]);
+  };
+
+  /* Moving focus over to next row containing searched word */
+  const handlePrevSearchItem = (): void => {
+    const oldIndex = searchedWordIndexes.indexOf(rowInFocus);
+    const adjustedSearchCount = currentSearchItemCount - 1;
+    let temp = foundWordIndex;
+
+    if (oldIndex < 0) {
+      return null;
+    }
+
+    setCurrentSearchItemCount(adjustedSearchCount);
+    setFoundWordIndex(--temp);
+    scrollToRow(searchedWordIndexes[--temp]);
+  };
+
+  const toolbarItems = (
+    <>
+      <ToolbarItem>
+        <ToolbarGroup>
+          <SearchInput
+            placeholder="Search"
+            value={searchedInput}
+            onNextClick={() => handleNextSearchItem()}
+            onPreviousClick={() => handlePrevSearchItem()}
+            onClear={() => handleClear()}
+            onChange={input => setSearchedInput(input)}
+            resultsCount={`${currentSearchItemCount} / ${searchedWordIndexes.length}`}
+            className={styles.toolbarSearchbar}
+          />
+        </ToolbarGroup>
+      </ToolbarItem>
+      {customControls && <ToolbarItem>{customControls}</ToolbarItem>}
```

Comment:
Ideally, I think if all your controls should be `ToolbarItems` then the wrapping of the controls should be done in the `CustomControl` component